### PR TITLE
s3: Adding RackCorp Object Storage to S3 definitions

### PR DESCRIPTION
**Pull request by bbabich**
#### Checklist

- [Y] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [Y] I have added tests for all changes in this PR if appropriate.
- [Y] I have added documentation for the changes if appropriate.
- [Y] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ Y] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: docs/content/s3.md [hunk 1]
```diff
@@ -2249,6 +2250,72 @@ So once set up, for example to copy files into a bucket
 rclone copy /path/to/files minio:bucket
 ```
 
+### RackCorp {#RackCorp}
+
+[RackCorp Object Storage](https://www.rackcorp.com/storage/s3storage) is an S3 compatible object storage platform from your friendly cloud provider RackCorp. The service is fast, reliable, well priced and located in many strategic locations unserviced by others, to ensure you can maintain data sovereignty.
+
+Before you can use RackCorp Object Storage, you'll need to "[sign up](https://www.rackcorp.com/signup)" for an account on our "[portal](https://portal.rackcorp.com)", where you can create an `access key`, a `secret key` and `buckets`, in your location of choice with ease. These details are required for the next steps of configuration, when `rclone config` asks for your `access_key_id` and `secret_access_key`.
+
+At present, the following `region` and `location_constraint` combinations are available. They are only relevant at the time of creating a bucket and will indicate the location or region where your data will be stored.
```
Comment (ncw) [inline docs/content/s3.md:2259]: Can you remove these lists of region and location_constraint? I see them getting out of date with respect to the actual configurator?

**Review by ncw — COMMENTED** [review]
This looks great - thanks!

See inline for small tweak :-)

**Comment by ivandeex** [comment]
@bbabich 
One last effort to get this merge - just one review comment to address...

File: docs/content/s3.md (hunk 1 above)
Comment (bbabich) [reply docs/content/s3.md]: Done @ncw; thanks for being so patient with me :)

**Comment by bbabich** [comment]
> needs rebasing on master to resolve merge conflicts review comments were provided 5 months ago and need addressing cc @bbabich

All fixed up @ivandeex :)

**Review by ncw — COMMENTED** [review]
This looks great - thanks.

Can you just `gofmt` the go file to remove the CI error then I'll merge - thanks :-)

**Comment by bbabich** [comment]
> This looks great - thanks.
> 
> Can you just `gofmt` the go file to remove the CI error then I'll merge - thanks :-)

Done @ncw

**Comment by ncw** [comment]
One more thing, sorry!

Yesterday I merged a patch which deals with the differences between the providers in a more sane way.

Can you add a `RackCorp` case to `setQuirks` and work out what quirks (if any) RackCorp needs? You can test with the integration tests (use `go test -v -remote rackcorp:` in the backend/s3 directory).

https://github.com/rclone/rclone/blob/25366268fe921608ee68c6dc33b744b80d860abe/backend/s3/s3.go#L1692-L1698

Thank you :-)

**Comment by bbabich** [comment]
> This looks great - thanks.
> 
> Can you just `gofmt` the go file to remove the CI error then I'll merge - thanks :-)

Done @ncw 

> One more thing, sorry!
> 
> Yesterday I merged a patch which deals with the differences between the providers in a more sane way.
> 
> Can you rebase this on master, then add a `RackCorp` case to `setQuirks` and work out what quirks (if any) RackCorp needs? You can test with the integration tests (use `go test -v -remote rackcorp:` in the backend/s3 directory).
> 
> https://github.com/rclone/rclone/blob/25366268fe921608ee68c6dc33b744b80d860abe/backend/s3/s3.go#L1692-L1698
> 
> Thank you :-)

This one has been done @ncw :)

File: docs/content/s3.md [hunk 2]
```diff
@@ -2346,6 +2347,27 @@ So once set up, for example, to copy files into a bucket
 rclone copy /path/to/files minio:bucket
 ```
 
+### RackCorp {#RackCorp}
+
+[RackCorp Object Storage](https://www.rackcorp.com/storage/s3storage) is an S3 compatible object storage platform from your friendly cloud provider RackCorp. The service is fast, reliable, well priced and located in many strategic locations unserviced by others, to ensure you can maintain data sovereignty.
```
Comment (ncw) [inline docs/content/s3.md:2352]: Can you hard wrap these docs to 80 characters please? We do the docs like that so they can be views in the terminal tool

**Review by ncw — COMMENTED** [review]
I was just about to merge this but I noticed a couple of very small things which need fixing first - see inline - sorry for not noticing these before.

Can you fix those and then rebase on master and squash to a single commit please then I'll merge.

Thank you :-)

**Review by ncw — APPROVED** [review]
That looks great now - thank you :-)

I'll merge it now.

**Pull request merged**